### PR TITLE
rpcserver+lncli: fix custom channel data encoding issue

### PR DESCRIPTION
**Pull request by guggero**
Fixes an error that occurred in `lncli listchannels` for non-custom channels.

### CODE REVIEW

File: cmd/commands/commands.go [hunk 1]
```diff
@@ -78,7 +78,9 @@ func replaceCustomData(jsonBytes []byte) ([]byte, error) {
 	var buf bytes.Buffer
 	err := json.Indent(&buf, replacedBytes, "", "    ")
 	if err != nil {
-		return nil, err
+		// If we can't indent the JSON, it likely means the replacement
+		// data wasn't correct, so we return the original JSON.
+		return jsonBytes, nil
```
Comment (dstadulis) [inline cmd/commands/commands.go:83]: Reviewer note: the jsonBytes with error context is decorated here:
https://github.com/lightningnetwork/lnd/blob/19653e1deca8baaa2c7f9bff2cef334dfc081eb9/cmd/commands/commands.go#L123C1-L126C32

File: rpcserver.go [hunk 2]
```diff
@@ -4606,6 +4606,11 @@ func encodeCustomChanData(lnChan *channeldb.OpenChannel) ([]byte, error) {
 	customOpenChanData := lnChan.CustomBlob.UnwrapOr(nil)
 	customLocalCommitData := lnChan.LocalCommitment.CustomBlob.UnwrapOr(nil)
 
+	// Don't write any custom data if both blobs are empty.
+	if len(customOpenChanData) == 0 && len(customLocalCommitData) == 0 {
```
Comment (dstadulis) [inline rpcserver.go:4610]: :+1:

**Review by dstadulis — APPROVED** [review]
utACK

**Review by ellemouton — APPROVED** [review]
Noice

File: cmd/commands/commands.go (hunk 1 above)
Comment (guggero) [reply cmd/commands/commands.go]: Hmm, good point. Updated.

**Pull request merged**